app/controllers/recipes_controller.rb

```diff
@@ -0,0 +1,42 @@
+
+class RecipesController < ApplicationController
+  around_action :catch_api_error
+
+  def root
+  end
+
+  def index
+    @query = params[:query]
+
+    @recipes = RecipeSearchWrapper.search_recipes(@query)
+    if @recipes.empty?
+      flash[:status] = :failure
+      flash[:result_text] = "There are no recipes for #{params[:query]}. Maybe you could write one?"
+      redirect_to root_path
+    else
+      @recipes = @recipes.paginate(:page => params[:page], :per_page => 10)
+    end
+  end
+
+  def show
+    @recipe = RecipeSearchWrapper.find_recipe(params[:uri])
+
+    unless @recipe
+      head :not_found
```

Comment:
Here you look for a `nil` recipe, but in your lib file you raise an exception if the API didn't return a result. This disconnect is why mangling the URI breaks the site.